src/pkg/cli/client/byoc/aws/byoc.go

```diff
@@ -322,81 +322,99 @@ func (b *ByocAws) findZone(ctx context.Context, domain, roleARN string) (string,
-		var delegationSetAlreadyReusable *r53types.DelegationSetAlreadyReusable
-		if errors.As(err, &delegationSetAlreadyCreated) || errors.As(err, &delegationSetAlreadyReusable) {
+
+		// Case 2b or 2c: The zone was not created by an older version of this CLI. We'll get the delegation set and let CD/Pulumi create/update the hosted zone
+		// TODO: we need to detect the case 2c where the zone was created by another tool and we need to use the existing NS records
+
+		// Create a reusable delegation set for the existing subdomain zone
+		delegationSet, err = aws.CreateDelegationSet(ctx, zone.Id, r53Client)
+		if delegationSetAlreadyReusable := new(r53types.DelegationSetAlreadyReusable); errors.As(err, &delegationSetAlreadyReusable) {
 			term.Debug("Route53 delegation set already created:", err)
-			delegationSet, err = aws.GetDelegationSet(ctx, r53Client)
+			delegationSet, err = aws.GetDelegationSetByZone(ctx, zone.Id, r53Client)
 		}
 		if err != nil {
-			return nil, AnnotateAwsError(err)
-		}
-		if len(delegationSet.NameServers) == 0 {
-			return nil, errors.New("no NS records found for the delegation set") // should not happen
-		}
-		term.Debug("Route53 delegation set ID:", *delegationSet.Id)
-		resp.DelegationSetId = strings.TrimPrefix(*delegationSet.Id, "/delegationset/")
-
-		// Ensure the NS records match the ones from the delegation set if the zone already exists
-		if zoneId != nil {
-			sort.Strings(nsServers)
-			sort.Strings(delegationSet.NameServers)
-			if !slices.Equal(delegationSet.NameServers, nsServers) {
-				track.Evt("Compose-Up delegateSubdomain diff", track.P("fromDS", delegationSet.NameServers), track.P("fromZone", nsServers))
-				term.Debugf("NS records for the existing subdomain zone do not match the delegation set: %v <> %v", delegationSet.NameServers, nsServers)
-			}
+			return nil, "", err
 		}
+	}
 
-		nsServers = delegationSet.NameServers
-	} else {
-		// Case 2a: The zone was created by the older CLI, we'll use the existing NS records; track how many times this happens
-		track.Evt("Compose-Up delegateSubdomain old", track.P("domain", projectDomain))
+	if len(delegationSet.NameServers) == 0 {
+		return nil, "", errors.New("no NS records found for the delegation set") // should not happen
 	}
-	resp.NameServers = nsServers
+	term.Debug("Route53 delegation set ID:", *delegationSet.Id)
+	delegationSetId = strings.TrimPrefix(*delegationSet.Id, "/delegationset/")
 
-	return &resp, nil
+	// Ensure the NS records match the ones from the delegation set if the zone already exists
+	sort.Strings(nsServers)
+	sort.Strings(delegationSet.NameServers)
+	if !slices.Equal(delegationSet.NameServers, nsServers) {
+		track.Evt("Compose-Up delegateSubdomain diff", track.P("fromDS", delegationSet.NameServers), track.P("fromZone", nsServers))
+		term.Debugf("NS records for the existing subdomain zone do not match the delegation set: %v <> %v", delegationSet.NameServers, nsServers)
+		// FIXME: this occurred 4 times
```

Comment:
Where are we with this FIXME comment? When you say "this occurred 4 times", which time period are we talking about?

Reply:
I just checked mix panel. I think it happens a total of 4 times over 6 mos. It's unexpected, so I need to dig in, but not related to the issue at hand.

Reply:
```suggestion
		// FIXME: this occurred 4 times between July 2024 and Jan 2025
```

Reply:
Actually, this PR fixed the issue, by using the new `GetDelegationSetByZone` to ensure the delegation set and the zone have the same name servers. Before, it would just do `GetDelegationSet` and you'd get a random one.

Reply:
So does that mean we should get rid of this comment then?

Reply:
Oh I must have had the wrong view open. I see it now.